Marlin/src/inc/SanityCheck.h

```diff
@@ -2873,6 +2873,29 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
   #endif
 #endif
 
+/**
+ * Screen Timeout feature not supported by these common displays
+*/
+#if HAS_SCREEN_TIMEOUT
+  #if HAS_MARLINUI_HD44780
+    #error "SCREEN_TIMEOUT feature not supported by HD44780 Reprap Discount Smart Controller 2004 LCD"
+  #elif ENABLED(REPRAPWORLD_GRAPHICAL_LCD)
+    #error "SCREEN_TIMEOUT feature not supported by REPRAPWORLD_GRAPHICAL_LCD"
+  #elif IS_U8GLIB_ST7920
+    #error "SCREEN_TIMEOUT feature not supported by Reprap Discount Full Graphics Smart Controller using ST7920"
+  #elif ENABLED(CARTESIO_UI)
+    #error "SCREEN_TIMEOUT feature not supported by CARTESIO_UI LCD"
+  #elif IS_U8GLIB_LM6059_AF
```

Comment:
these devices support sleep
https://github.com/MarlinFirmware/U8glib-HAL/blob/master/src/clib/u8g_dev_st7565_dogm128.c#L171-L176
https://github.com/MarlinFirmware/U8glib-HAL/blob/master/src/clib/u8g_dev_st7565_lm6059.c#L144-L149
https://github.com/MarlinFirmware/U8glib-HAL/blob/master/src/clib/u8g_dev_st7565_64128n.c#L185-L190

Reply:
Should I add these displays to the updates at borland1/Marlin#1 ?

Reply:
it is ok to add

Reply:
By the way, If you don't mind, could you merge u8glib-hal for supporting additional devices?